Makefile

```diff
@@ -4,7 +4,12 @@
 PROJECT_NAME ?= upjet-provider-template
 PROJECT_REPO ?= github.com/upbound/$(PROJECT_NAME)
 
-export TERRAFORM_VERSION ?= 1.2.1
+export TERRAFORM_VERSION ?= 1.5.7
+
+# Do not allow a version of terraform greater than 1.5.x, due to versions 1.6+ being
+# licensed under BSL, which is not permitted.
+export TERRAFORM_VERSION_CEILING ?= 1.6
+TERRAFORM_VERSION_VALID := $(shell [ $(TERRAFORM_VERSION) = `echo "$(TERRAFORM_VERSION)\n$(TERRAFORM_VERSION_CEILING)" | sort -V | head -n1` ] && echo 1 || echo 0)
```

Comment:
also a bit worried about the portability of using control chars (`\n`) with `echo` - that isn't allowed as per https://www.shellcheck.net/
```
^-- [SC2028](https://www.shellcheck.net/wiki/SC2028) (info): echo may not expand escape sequences. Use printf.
```

Will look into using `printf` for this check instead 😇

Reply:
i have incorporated `printf` now instead of using `echo` and have tested the same scenarios are still working.